### PR TITLE
#56 - Tests on Windows enabled, upgraded to ASTO to 0.16

### LINKED ISSUE #56: AstoBlobsITCase.java:43-45: Fix a bug with RxFile which...
The puzzle `43-b26c76b8` from #43 has to be resolved: 

https://github.com/artipie/docker-adapter/blob/0bafcf84f9271a298940b9719b9a2e71ad138cb3/src/test/java/com/artipie/docker/asto/AstoBlobsITCase.java#L43-L45

The puzzle was created by @g4s8 on 12-Feb-20. 

Estimate: 30 minutes,  role: DEV. 

If you have any technical questions, don't ask me, submit new tickets instead. The task will be \"done\" when the problem is fixed and the text of the puzzle is _removed_ from the source code. Here is more about [PDD](http://www.yegor256.com/2009/03/04/pdd.html) and [about me](http://www.yegor256.com/2017/04/05/pdd-in-action.html).

### PR DESCRIPTION
**Pull request by olegmoz**
Closes #56 
Upgraded ASTO to 0.16
Replaced FileStorage usage with InMemoryStorage in tests
Enabled tests on Windows

### CODE REVIEW

**Comment by 0crat** [comment]
This pull request [#61](https://github.com/artipie/docker-adapter/issues/61) is assigned to @vatavuk[/z](https://www.0crat.com/u/vatavuk), here is [why](https://www.0crat.com/footprint/CT2E6TK9B/0a4e93e0-7af8-41d9-8e3a-59e0ac01fb56); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @g4s8[/z](https://www.0crat.com/u/g4s8) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; there will be a monetary reward for this job

File: src/main/java/com/artipie/docker/asto/AstoBlobs.java [hunk 1]
```diff
@@ -121,12 +121,18 @@ public AstoBlobs(final Storage asto) {
...
             .doOnTerminate(out::close)
             .flatMap(
-                digest -> SingleInterop.fromFuture(
-                    this.asto.save(
-                        new Key.From(RegistryRoot.V2, new BlobRef(digest).string(), "data"),
-                        FlowAdapters.toFlowPublisher(new RxFile(tmp).flow())
-                    ).thenApply(none -> digest)
-                )
+                digest -> {
+                    final Vertx vertx = Vertx.vertx();
```
Comment (g4s8) [inline src/main/java/com/artipie/docker/asto/AstoBlobs.java]: @olegmoz this call creates a thread pool for IO operations. I'd move it `private static final Vertx` field, or maybe `private static final Filesystem`.

Comment (olegmoz) [reply src/main/java/com/artipie/docker/asto/AstoBlobs.java]: @g4s8 done. Although I think we should avoid using tmp file here later.

**Comment by 0crat** [comment]
Job was finished in 14 hours, bonus for fast delivery is possible (see [§36](http://www.zerocracy.com/policy.html#36))

**Comment by 0crat** [comment]
Job audit: `ARC` didn't merge or close PR, performer @vatavuk[/z](https://www.0crat.com/u/vatavuk)[/z](https://www.0crat.com/u/vatavuk) didn't make CR comments

**Pull request merged**